scripts/dts/edtlib.py

```diff
@@ -1659,26 +1659,37 @@ def _check(self, require_compatible, require_description):
                      f"field in {self.path} - "
                      f"should be a string, not {type(compatible).__name__}")
         elif require_compatible:
-            _err(f"missing 'compatible' property in {self.path}")
-
-        if "description" not in raw and require_description:
-            _err(f"missing 'description' property in {self.path}")
-
-        for prop in "title", "description":
-            if prop in raw and (not isinstance(raw[prop], str) or
-                                not raw[prop]):
-                _err(f"malformed or empty '{prop}' in {self.path}")
-
-        ok_top = {"title", "description", "compatible", "properties",
-                  "bus", "on-bus", "parent-bus", "child-bus", "parent", "child",
-                  "child-binding", "sub-node"}
-
-        for prop in raw:
-            if prop == "#cells": # clean error for users of legacy syntax
-                _err(f"malformed '{prop}:' in {self.path}, "
-                     "expected *-cells syntax")
-            if prop not in ok_top and not prop.endswith("-cells"):
-                _err(f"unknown key '{prop}' in {self.path}, "
+            _err(f"missing 'compatible' in {self.path}")
+
+        if "description" in raw:
+            description = raw["description"]
+            if not isinstance(description, str) or not description:
+                _err(f"malformed or empty 'description' in {self.path}")
+        elif require_description:
+            _err(f"missing 'description' in {self.path}")
+
+        # Allowed top-level keys. The 'include' key should have been
+        # removed by _load_raw() already.
+        ok_top = {"description", "compatible", "bus", "on-bus",
+                  "properties", "child-binding"}
+
+        # Descriptive errors for legacy bindings.
+        legacy_errors = {
```

Comment:
Do we think its time to remove these?

Reply:
> Do we think its time to remove these?

I'm actually adding more in this PR because the errors were never there before. If you look at what `ok_top` is in master, you can see that keys like `parent` are allowed but silently ignored. This PR stops allowing them for the first time, so adding new error messages makes sense to me.